tests/processors/base/writers/test_trial_and_record.py

```diff
@@ -284,16 +284,23 @@ def test_writes_record_with_valid_url(self, conn, trial, nct_source):
 
         assert conn['database']['records'].count(id=record_id) == 1
 
-    def test_doesnt_update_None_parameters(self, conn, trial, record):
+    def test_doesnt_alter_record_parameters(self, conn, trial, record):
```

Comment:
I don't understand what we're testing here. It feels like the objective is to guarantee that we don't set a record's `is_primary` to `False`, and `source_url` and `source_id` to `None`. However, the actual test is that `source_url` is a valid URL. Because if you called `_write_record(conn_mock, trial_attrs, record, None, 'invalid_source_url', False)`, the exception would be different (some `sqlalchemy` exception I imagine).

The deeper problem here IMHO is trying to test the behaviour of the code, not the results of running it. In other words, this is a white box test, where a black box test is usually better. This way, the tests don't break unnecessarily if we refactor the code.

My suggestion would be to simply delete this test. I think the only param type validation that we aren't testing is checking that `source_id` isn't `None` (we already check if `source_url` is valid and `is_primary` can be `None`). However, what guarantees `source_id` can't be `None` is the database. As it doesn't change often, I don't think we need this test.